sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -192,12 +198,20 @@ case class CreateViewCommand(
       // package (e.g., HiveGenericUDF).
       def verify(child: LogicalPlan) {
         child.collect {
-          // Disallow creating permanent views based on temporary views.
+          // Permanent views will be created as temporary view if based on temporary view.
           case UnresolvedRelation(AsTableIdentifier(ident))
             if sparkSession.sessionState.catalog.isTemporaryTable(ident) =>
-            // temporary views are only stored in the session catalog
-            throw new AnalysisException(s"Not allowed to create a permanent view $name by " +
-              s"referencing a temporary view $ident")
```

Comment:
We can improve the error message and suggest end users to add the keyword TEMPORARY to create a temporary view instead.

Reply:
PR for above suggestion https://github.com/apache/spark/pull/26731

Reply:
@gatorsmile also review https://github.com/apache/spark/pull/26712